### PR TITLE
Build script uses clingo

### PR DESCRIPTION
**Pull request by bvanessen**
This PR simplifies the build script and gives directions on how the user needs to setup the clingo concretizer.  Using the clingo concretizer greatly simplifies what information the user has to specify to get spack to find a valid solution.

### CODE REVIEW

File: docs/building_lbann.rst [hunk 1]
```diff
@@ -39,8 +39,30 @@ Setup Spack
...
+
+       emacs -nw <path to installation>/spack.git/etc/spack/config.yaml
+
+    Ensure the `config.yaml` file contains the following lines:
+
+    .. code-block:: bash
+
+       config:
+         concretizer: clingo
+
```
Comment (benson31) [inline docs/building_lbann.rst:64]: We don't need to tell folks how to edit files. If they don't already know, they need to read a "linux command line 101" primer first, and that's not our job.
```suggestion
    Create the file `${SPACK_ROOT}/etc/spack/config.yaml` if it doesn't exist. Verify that this file contains the following lines, and add them if necessary:

    .. code-block:: bash

       config:
         concretizer: clingo

```

**Review by mrwyattii — APPROVED** [review]
LGTM, tested and working

**Review by benson31 — APPROVED** [review]
LGTM. Built successfully on Pascal.

**Comment by bvanessen** [comment]
@samadejacobs @timmoon10 I have updated the script and instructions to automatically switch your spack repository to use clingo.  This simplifies the developer's instructions.

**Review by timmoon10 — APPROVED** [review]
When I uncomment the code that sets `concretization: together`, this works well for me. I was able to install LBANN and PyTorch together without fussing about getting the dependencies to match.

File: scripts/build_lbann.sh [hunk 2]
```diff
@@ -333,6 +390,11 @@ if [[ -n "${INSTALL_DEPS:-}" ]]; then
     echo ${CMD} | tee -a ${LOG}
     [[ -z "${DRY_RUN:-}" ]] && { ${CMD} || exit_on_failure "${CMD}"; }
 
+#     SPACK_ENV_YAML_FILE="${SPACK_ROOT}/var/spack/environments/${LBANN_ENV}/spack.yaml"
+# cat <<EOF  >> ${SPACK_ENV_YAML_FILE}
+#   concretization: together
+# EOF
+
```
[the remark this answers is not in the document]
Comment (bvanessen) [reply scripts/build_lbann.sh:397]: Good to hear.  Now we just need the relevant spack PR merged.

**Pull request merged**